lib/db/set.go

```diff
@@ -123,6 +131,35 @@ func (s *FileSet) recalcCounts() error {
 	return s.meta.toDB(s.db, []byte(s.folder))
 }
 
+// Verify the local sequence number from actual sequence entries. Returns
+// true if it was all good, or false if a fixup was necessary.
+func (s *FileSet) verifyLocalSequence() (ok bool) {
+	// Walk the sequence index from the current (supposedly) highest
+	// sequence number and raise the alarm if we get anything. This recovers
+	// from the occasion where we have written sequence entries to disk but
+	// not yet written new metadata to disk.
+	//
+	// Note that we can have the same thing happen for remote devices but
+	// there it's not a problem -- we'll simply advertise a lower sequence
+	// number than we've actually seen and receive some duplicate updates
+	// and then be in sync again.
+
+	curSeq := s.meta.Sequence(protocol.LocalDeviceID)
+	snap := s.Snapshot()
+	ok = true
```

Comment:
You could just drop the named return argument, in which case that couldn't have gone forgotten before